frontend/src/components/Graph.tsx

```diff
@@ -57,41 +65,40 @@ function Graph() {
 
   return (
     <div>
-      {knownData.length == 0 ? (
-        <h1>loading</h1>
+      {knownData.length === 0 ? (
+        <h1>Loading...</h1>
       ) : (
-        <div>
-          <LineChart width={800} height={500}>
-            <CartesianGrid strokeDasharray="3 3" />
-            <XAxis
-              angle={-45}
-              textAnchor="end"
-              dataKey="date"
-              scale="time"
-              type="number"
-              tickFormatter={dateFormatter}
-              domain={[
-                knownData[0].data[0].date.getTime(), // gets the first date across all data
-                knownData.slice(-1)[0].data.slice(-1)[0].date.getTime(), // gets the last date across all data
-              ]}
+        <LineChart width={800} height={500}>
+          <CartesianGrid strokeDasharray="3 3" />
+          <XAxis
+            angle={-45}
+            textAnchor="end"
+            dataKey="date"
+            scale="time"
+            type="number"
+            tickFormatter={dateFormatter}
+            domain={[
+              knownData[0].data[0].date.getTime(),
+              knownData[0].data[knownData[0].data.length - 1].date.getTime(),
```

Comment:
line 82 wont work correctly.

it should be using the last index of knownData

Reply:
fixed it I think